test/test_autograd.py

```diff
@@ -3968,9 +3968,11 @@ def test_gradcheck_sparse_input(self):
         def fn(sparse):
             return torch.sparse.sum(sparse)
 
-        gradcheck(fn, torch.rand(10).to_sparse().requires_grad_(True), check_sparse_nnz=True)
+        gradcheck(fn, torch.rand(10).to_sparse().requires_grad_(True),
+                  check_sparse_nnz=True, check_batched_grad=False)
```

Comment:
Why is the batched grad not supported here anymore?

Reply:
Note an interesting failure was happening here before. Calling `tensor.coalesce()` on an already-coalesced tensor sets its `grad_fn`, making it no-longer an autograd leaf. When the batched grad check then calls `torch.autograd.grad` here:
https://github.com/pytorch/pytorch/blob/97568d7471bc3ae1eac6a88c46b1ae96c60cc334/torch/autograd/gradcheck.py#L247-L250
It doesn't recognize the inputs as leafs of the graph and the result ends up being a dense tensor of all zeros (because of `allow_unused=True`). Fixing this bug makes the test run properly, and it fails because sparse tensors don't work well inside batched tensors.

Reply:
Specifically, `batchedTensorForLoopFallback` calls `flat_output.view(output_sizes)` on the output which isn't implemented for sparse tensors.
https://github.com/pytorch/pytorch/blob/b8e6e2971cba779daf14652d05f427544ab635b5/aten/src/ATen/BatchedFallback.cpp#L370

Reply:
cc @zou3519

Reply:
I'm OK with the regression in batched grad coverage here, we are not going to try to handle sparse tensors in the near future.